winlogbeat/checkpoint/checkpoint.go

```diff
@@ -89,6 +89,12 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 		}
 	}
 
+	// Write the state file to verify we have have permissions.
+	err = c.flush()
```

Comment:
You are absolutely correct that it is not necessary to actually write, it just needs to open for write to do this test. I just figured I would call `flush` because it takes care of

- verifying that the directory exists or can be created,
- verifying that the state file can be written, and
- verifying that the "`.new`" state file can be written.